src/Microsoft.Health.Fhir.Core/Models/ResourceChangeData.cs

```diff
@@ -0,0 +1,34 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Fhir.Core.Models
+{
+    public class ResourceChangeData
+    {
+        public ResourceChangeData(long id, DateTime timestamp, string resourceId, short resourceTypeId, int resourceVersion, byte resourceChangeTypeId)
```

Comment:
Long parameter list is code smell, let's make sure we have a backlog for improvement. Options could be the new csharp feature record or by adding a model builder pattern or something else.

Reply:
Created a user store [83431](https://microsofthealth.visualstudio.com/Health/_workitems/edit/83431) as a backlog.